src/dask_awkward/utils.py

```diff
@@ -149,3 +149,60 @@ def second(seq: Iterable[T]) -> T:
     the_iter = iter(seq)
     next(the_iter)
     return next(the_iter)
+
+
+def field_access_like(entry: Any) -> bool:
+    if isinstance(entry, str):
+        return True
+    if isinstance(entry, (list, tuple)) and all(isinstance(x, str) for x in entry):
+        return True
+    return False
+
+
+def field_access_to_front(seq: Sequence[Any]) -> tuple[tuple[Any, ...], int]:
+    """Move field access to the front of a sequence.
+
+    We have multiargument getitem calls we want to bring the field
+    access calls to the front. For example
+
+    >>> a[0, "foo"]
+
+    Is the same as
+
+    >>> a["foo", 0]
+
+    But the latter starts with something that is trivially
+    map-partitionable. This function helps us write out the logic for
+    getitem calls.
+
+    Parameters
+    ----------
+    seq : Sequence[Any]
+        Sequence to reorder.
+
+    Returns
+    -------
+    tuple[Any, ...]
+        Reordered sequence with field accesses brought to the front.
+    int
+        Total number of field accesses.
+
+    Examples
+    --------
+    >>> where = [0, ["foo", "bar"], "x"]
+    >>> new, n = field_access_to_front(where)
+    >>> new
+    [["foo", "bar"], "x", 0]
+    >>> n
+    2
+
+    """
+    new_seq: list[Any] = []
+    n_front = 0
+    for entry in seq:
+        if field_access_like(entry):
+            new_seq.insert(n_front, entry)
+            n_front += 1
+        else:
+            new_seq.append(entry)
+    return tuple(new_seq), n_front
```

Comment:
Can't we use sorted and/or itertools.groupby?
```suggestion
    reversed(seq, key=lambda x: not field_access_like(x))
```

Reply:
We could but the `n_front` variable is useful in this case for knowing _where_ we stop having field-access like elements in the list.

Reply:
That you could get from groupby (or map, filter, any...)
```suggestion
    return sorted(seq, key=field_access_like), sum(map(field_access_like)) # assuming field_access_like is inverted
```